mlflow/pipelines/steps/train.py

```diff
@@ -158,6 +172,104 @@ def _run(self, output_directory):
             raw_validation_df, prediction_result, self.target_col
         )
 
+        from mlflow.entities import ViewType
+
+        mlflow_client = MlflowClient()
+        exp_id = _get_experiment_id()
+
+        primary_metric_greater_is_better = self.metric_greater_is_better[self.primary_metric]
+
+        primary_metric_order = "DESC" if primary_metric_greater_is_better else "ASC"
+
+        search_max_results = 100
+        search_result = mlflow_client.search_runs(
+            experiment_ids=exp_id,
+            run_view_type=ViewType.ALL,
+            max_results=search_max_results,
+            order_by=[f"metrics.{self.primary_metric} {primary_metric_order}"],
```

Comment:
```suggestion
            order_by=[f"metrics.{self.primary_metric}_on_data_validation {primary_metric_order}"],
```

I think the order-by expression is missing a `_on_data_xxx` suffix.

Reply:
Good catch! Fixed!